pkgs/applications/networking/browsers/firefox/update.nix

```diff
@@ -0,0 +1,54 @@
+{ name
+, writeScript
+, xidel
+, coreutils
+, gnused
+, gnugrep
+, curl
+, ed
+, sourceSectionRegex ? "${name}-unwrapped = common"
+, basePath ? "pkgs/applications/networking/browsers/firefox"
+, baseUrl ? "http://archive.mozilla.org/pub/firefox/releases/"
+, versionSuffix ? ""
+}:
+
+let
+  version = (builtins.parseDrvName name).version;
+in writeScript "update-${name}" ''
+  PATH=${coreutils}/bin:${gnused}/bin:${gnugrep}/bin:${xidel}/bin:${curl}/bin:${ed}/bin
```

Comment:
This really should be abstracted over (stdenv-style). E.g.
```
  mkUpdateScript "update-${name}"
    [ xidel curl ]
    ''
      script...
    ''
```
where `mkUpdateScript` takes care of setting `PATH` and adding default dependencies like coreutils.